xbmc/Application.cpp

```diff
@@ -1990,17 +1990,17 @@ void CApplication::Render()
   else
     flip = true;
 
-  //fps limiter, make sure each frame lasts at least singleFrameTime milliseconds
-  if (limitFrames || !(flip || m_bPresentFrame))
+  //Set max fps for static frames
+  if (!limitFrames && !(flip || m_bPresentFrame))
   {
-    if (!limitFrames)
-      singleFrameTime = 40; //if not flipping, loop at 25 fps
-
-    unsigned int frameTime = now - m_lastFrameTime;
-    if (frameTime < singleFrameTime)
-      Sleep(singleFrameTime - frameTime);
+    singleFrameTime = 40; //if not flipping, loop at 25 fps
   }
 
+  //fps limiter, make sure each frame lasts at least singleFrameTime milliseconds
+  unsigned int frameTime = now - m_lastFrameTime;
+  if (frameTime < singleFrameTime)
+    Sleep(singleFrameTime - frameTime);
+
```

Comment:
this seems wrong. we don't want this pulled out of the above block. does your system goes above 100fps?